pkg/util/redact/redact.go

```diff
@@ -60,3 +66,105 @@ func (s redactStringer) String() string {
 func Stringer(mode string, input fmt.Stringer) redactStringer {
 	return redactStringer{mode, input}
 }
+
+func DeRedactFile(remove bool, input string, output string) error {
+	ifile, err := os.Open(input)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer ifile.Close()
+
+	var ofile io.Writer
+	if output == "-" {
+		ofile = os.Stdout
+	} else {
+		file, err := os.OpenFile(output, os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
Better consider about compressed files. Because tidb log will support it after https://github.com/pingcap/tidb/pull/51249

Reply:
Consider it later